src/pages/index.js

```diff
@@ -31,56 +81,73 @@ const popupEditProfileForm = new PopupWithForm(
   ".modal_profile-edit",
   handleProfileEditSubmit
 );
-const gallerySection = new Section(
-  { items: initialCards, renderer: cardRenderer },
-  cardGalleryEL
+const popupAvatarForm = new PopupWithForm(
+  ".modal_change-avatar",
+  handleAvatarFormSubmit
+);
+////////////////////////////////////////////////////////////
+///////////////////////   API    ///////////////////////////
+////////////////////////////////////////////////////////////
+const apiCall = new Api(
+  "https://around-api.en.tripleten-services.com/v1",
+  serverToken,
+  apiFunctionObj
 );
-
-//////////////////////////  functions  //////////////////////////
+////////////////////////////////////////////////////////////
+///////////////////////   FUNCTIONS    /////////////////////
+////////////////////////////////////////////////////////////
+function handleDeleteFormSubmit(data) {
+  data.cardElement.remove();
+  apiCall.deleteCard(data.data);
+}
+function handleAvatarFormSubmit(data) {
+  apiCall.patchAvatar(data);
```

Comment:
Every server request in `index.js` should look something like this:

```js
api.someMethod()
.then(res => {
// here is the response that should be used to update the DOM.
// also in `then` you should close the popup
})
.catch(err => // here you catch possible errors)
.finally(() => // here you bring the default button text back )
``` 

Please, correct it in the whole code